ovis_robotiq_gripper/nodes/ovis_gripper_control.py

```diff
@@ -0,0 +1,131 @@
+# "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+# LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+# FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+# COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+# INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+# BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+# LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+# CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+# LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+# ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+# POSSIBILITY OF SUCH DAMAGE.
+#
+# Copyright (c) 2012, Robotiq, Inc.
+# Revision $Id$
+
+# Base imports
+import sys
+import os
+
+# Message imports
+from ovis_robotiq_gripper.msg import _OvisGripperPosition as positionMsg
+from ovis_robotiq_gripper.msg import _OvisGripper_robot_output as outputMsg
+from ovis_robotiq_gripper.msg import _OvisGripper_robot_input as inputMsg
+import gripper_constants as cons
+
+# Custom, Robotiq and ROS imports
+import robotiq_modbus_rtu.comModbusRtu
+import ovis_robotiq_gripper.baseRobotiq2FGripper
+import rospy
+import roslib
+roslib.load_manifest('robotiq_2f_gripper_control')
+roslib.load_manifest('robotiq_modbus_rtu')
+
+# Global declararions
+command = outputMsg.OvisGripper_robot_output()
+current_position = inputMsg.OvisGripper_robot_input()
+
+
+def command_callback(input):
+    global command
+    global current_position
+
+    # Default values setup
+    command.rACT = cons.ON  # activates the gripper
+    command.rGTO = cons.ON  # Go to requested position
+    command.rFR = cons.MAX_FORCE  # maximum force
+    command.rSP = cons.MAX_SPEED
+
+    if input.position == 0:
+        command.rACT = cons.OFF  # activates the gripper
+        command.rGTO = cons.OFF  # Go to requested position
+        rospy.logdebug("Debug info: position = 2 and command sent is OFF")
+
+    if input.position == 1:
+        rospy.logdebug("Debug info: position = 1 and command sent is ON")
+
+    elif input.position == 2 and current_position.gPO <= 3:
+        command.rPR = cons.CLOSE  # close
+
+    elif input.position == 2:
+        command.rPR = cons.OPEN  # open
+
+def mainLoop(device):
+
+    # Gripper is a 2F with a TCP connection
+    gripper = ovis_robotiq_gripper.baseRobotiq2FGripper.robotiqbaseRobotiq2FGripper()
+    gripper.client = robotiq_modbus_rtu.comModbusRtu.communication()
+
+    # We connect to the address received as an argument
+    gripper.client.connectToDevice(device)
+
+    rospy.init_node("ovis_robotiq_gripper")
+
+    # The Gripper status is published on the topic named 'gripper_input'
+    pubInput = rospy.Publisher(
+        'gripper_input', inputMsg.OvisGripper_robot_input, queue_size=1)
+
+    # The Gripper command is published on the topic named 'gripper_output'
+    pubOutput = rospy.Publisher(
```

Comment:
Same thing here, I'm not sure why we are using `out` instead of `command` or `cmd`?